lib/auth/init.go

```diff
@@ -1020,6 +1028,57 @@ func migrateCertAuthorities(ctx context.Context, asrv *Server) error {
 	return nil
 }
 
+// migrateDBAuthority copies Host CA as Database CA. Before v9.0 database access was using host CA to sign all
```

Comment:
I'm thinking more about this and now pondering whether this is a good approach from the security standpoint. The scenario I've in my mind is this: say your host CA gets compromised, you rotate it but aren't aware that your db CA after this migration is the same and that you need to rotate it too. In a situation like this, you would probably rotate all your CAs anyway but I wonder if this may not be a desired outcome following the principle of least surprise.

@fspmarshall @russjones Do you guys have any thoughts about this migration approach versus just leaving database CA empty and falling back to host CA and letting users explicitly initialize their database CA by performing rotation? We considered both and chose this one for several reasons (e.g. we'd likely need to update rotation logic to support empty CAs for the other one, instruct users to perform rotation after upgrade if they want to migrate to separate database CA, etc.) but now wondering if this might not be the best way.

Reply:
@r0mant I find the scenario you described pretty convincing.  If a user wants a CA gone, we should protect them from accidentally leaving it in circulation like that.

If updating rotation logic to support empty CAs is tricky, what about updating it to check for duplicate CAs and rotate all CAs that match the targeted CA?  E.g. if I try to rotate my host CA, and teleport notices that the DB ca happens to be equivalent, could it just automatically rotate it in lockstep?  Injecting an additional CA type [here](https://github.com/gravitational/teleport/blob/361d78d89ec33b46ac58dbb3492b1e45e4b7e6b8/lib/auth/rotate.go#L213) would be fairly straightforward.

While doing that kind of thing silently is generally sketchy, I think we can safely assume that we won't have matching CAs by chance, meaning that it will basically always be a case like this one; we've split the duties that one CA used to perform into two or more separate CAs, and are now rotating for the first time since that split.

Reply:
Makes sense. @jakule Could you please take a look which approach looks more straightforward to implement (supporting empty CA vs rotating all matching CAs in lockstep) and let's do it.

Reply:
@r0mant I'll look into that. Right now I'm leaning towards rotating all matching certificates. Fallback is probably simpler to implement, but if we decide to go this route, I don't think we will be able to remove that logic anytime soon, as we don't know when everyone decide to rotate their certs. Rotating all matching certs keeps the logic in one place and allows us to reuse it if we ever need it (Desktop Access for example).